src/block/suite.rs

```diff
@@ -0,0 +1,178 @@
+use header::{SuiteLabel, SuiteHeader};
+use block::Context;
+
+/// Test suites bundle a set of closely related test examples into a logical execution group.
+pub struct Suite<T> {
+    pub(crate) header: SuiteHeader,
+    pub(crate) environment: T,
+    pub(crate) context: Context<T>,
+}
+
+impl<T> Suite<T> {
+    pub(crate) fn new(header: SuiteHeader, environment: T, context: Context<T>) -> Self {
+        Suite {
+            header: header,
+            environment: environment,
+            context: context,
+        }
+    }
+
+    pub fn num_blocks(&self) -> usize {
+        self.context.num_blocks()
+    }
+
+    pub fn num_examples(&self) -> usize {
+        self.context.num_examples()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.context.is_empty()
+    }
+}
+
+unsafe impl<T> Send for Suite<T>
```

Comment:
move the Send and Sync impl after the documented block code

Reply:
Sure? `Send` and `Sync` belong to `Suite<T>`. The documented code block contains free functions.